models/files/pre.R

```diff
@@ -0,0 +1,178 @@
+modelInfo <- list(
+  library = "pre",
+  type = c("Classification", "Regression"),
+  parameters = data.frame(parameter = c("sampfrac", "maxdepth", 
+                                        "learnrate", "mtry", 
+                                        "ntrees", "use.grad", 
+                                        "penalty.par.val"),
+                          class = c(rep("numeric", times = 5), 
+                                    "logical", "character"),
+                          label = c("Subsampling Fraction", 
+                                    "Max Tree Depth", 
+                                    "Shrinkage", 
+                                    "# Randomly Selected Predictors",
+                                    "# Trees", 
+                                    "Employ Gradient Boosting", 
+                                    "Regularization Parameter")),
+  grid = function(x, y, len = NULL, search = "grid", 
+                  sampfrac = .5, maxdepth = 3L, learnrate = .01, 
+                  mtry = Inf, ntrees = 500, use.grad = TRUE, 
+                  penalty.par.val = "lambda.1se") {
+    if (search == "grid") {
+      if (!is.null(len)) {
+        maxdepth <- c(3L, 4L, 2L, 5L, 1L, 6:len)[1:len] 
+        if (len > 2) {
+          sampfrac <- c(.5, .75, 1)
+        }
+        if (len > 1) {
+          penalty.par.val = c("lambda.min", "lambda.1se")
+        }
+      }
+      out <- expand.grid(sampfrac = sampfrac, maxdepth = maxdepth, 
+                         learnrate = learnrate, mtry = mtry, 
+                         ntrees = ntrees, use.grad = use.grad,  
+                         penalty.par.val = penalty.par.val)
+    } else if (search == "random") {
+      out <- data.frame(
+        sampfrac = sample(c(.5, .75, 1), size = len, replace = TRUE),
+        maxdepth = sample(2L:6L, size = len, replace = TRUE), 
+        learnrate = sample(c(0.001, 0.01, 0.1), size = len, replace = TRUE),
+        mtry = sample(c(ceiling(sqrt(ncol(x))), ceiling(ncol(x)/3), ncol(x)), size = len, replace = TRUE),
+        ntrees = rep(500, times = len),
+        use.grad = sample(c(TRUE, FALSE), size = len, replace = TRUE),
+        penalty.par.val = sample(c("lambda.1se", "lambda.min"), size = len, replace = TRUE))
+    }
+    return(out)
+  },
+  fit = function(x, y, wts = NULL, param, lev = NULL, last = NULL, 
```

Comment:
`family` can be passed in via `...` but you should probably intercept the `...` and set `family` if it is not passed in. See [the `glm` code](https://github.com/topepo/caret/blob/1cd9e2283e48a99828e6695f0cdbcb0794b08e10/models/files/glm.R#L14) for an example. 

It's a it more work but most of the other modules do so and it would cut down on a lot of SO questions and issues filed.

Reply:
Dots are intercepted now and family argument set if not passed to dots.